core/src/ledger/eth_bridge/storage/whitelist.rs

```diff
@@ -0,0 +1,164 @@
+//! ERC20 token whitelist storage data.
+//!
+//! These storage keys should only ever be written to by governance,
+//! or `InitChain`.
+
+use std::str::FromStr;
+
+use super::super::ADDRESS as BRIDGE_ADDRESS;
+use super::{prefix as ethbridge_key_prefix, wrapped_erc20s};
+use crate::types::ethereum_events::EthAddress;
+use crate::types::storage;
+use crate::types::storage::DbKeySeg;
+use crate::types::token::{denom_key, minted_balance_key};
+
+mod segments {
+    //! Storage key segments under the token whitelist.
+    use namada_macros::StorageKeys;
+
+    use crate::types::address::Address;
+    use crate::types::storage::{DbKeySeg, Key};
+
+    /// The name of the main storage segment.
+    pub(super) const MAIN_SEGMENT: &str = "whitelist";
+
+    /// Storage key segments under the token whitelist.
+    #[derive(StorageKeys)]
+    pub(super) struct Segments {
+        /// Whether an ERC20 asset is whitelisted or not.
+        pub whitelisted: &'static str,
+        /// The token cap of an ERC20 asset.
+        pub cap: &'static str,
+    }
+
+    /// All the values of the generated [`Segments`].
+    pub(super) const VALUES: Segments = Segments::VALUES;
+
+    /// Listing of each of the generated [`Segments`].
+    pub(super) const ALL: &[&str] = Segments::ALL;
```

Comment:
that's a nice way to organize these and to make the macro generated stuff visible